services-directions-models/src/main/java/com/mapbox/api/directions/v5/utils/FormatUtils.java

```diff
@@ -0,0 +1,243 @@
+package com.mapbox.api.directions.v5.utils;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.mapbox.geojson.Point;
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Methods to convert models to Strings.
+ */
+public class FormatUtils {
+  /**
+   * Returns a string containing the tokens joined by delimiters. Doesn't remove trailing nulls.
+   *
+   * @param delimiter the delimiter on which to split.
+   * @param tokens A list of objects to be joined. Strings will be formed from the objects by
+   *               calling object.toString().
+   * @return {@link String}
+   */
+  @Nullable
+  public static String join(@NonNull CharSequence delimiter, @Nullable List<?> tokens) {
+    return join(delimiter, tokens, false);
+  }
+
+  /**
+   * Returns a string containing the tokens joined by delimiters.
+   *
+   * @param delimiter the delimiter on which to split.
+   * @param tokens A list of objects to be joined. Strings will be formed from the objects by
+   *               calling object.toString().
+   * @param removeTrailingNulls true if trailing nulls should be removed.
+   * @return {@link String}
+   */
+  @Nullable
+  public static String join(@NonNull CharSequence delimiter, @Nullable List<?> tokens,
+      boolean removeTrailingNulls) {
```

Comment:
is `removeTrailingNulls` always `false`? Maybe we can remove it if that's the case? One less thing to test and maintain.

Reply:
it is true for `waypointIndices`
```
waypointIndices(FormatUtils.join(";", waypointIndices, true));
```

Reply:
Do we need it though? The docs are clear that we are looking for indices and passing `null`s will be invalid, and the response error will point it out probably. Removing the values might just introduce unexpected behavior.

Reply:
I added it just to try to handle customers invalid input String.
But maybe you are right and we should not do it under the hood.